packages/driver/src/cypress.ts

```diff
@@ -530,6 +530,11 @@ class $Cypress {
         return this.emit('form:submitted', args[0])
 
       case 'app:window:load':
+        this.emit('internal:window:load', {
+          type: 'same:domain',
+          event: args[0],
+        })
+
         return this.emit('window:load', args[0])
```

Comment:
Could be my lack of previous reviews on this new API, but why is this new event needed in addition to `window:load`? Or was it meant to replace `window:load`?

If there is page navigation  `window.history.go()` or `cy.go()` between the original & previous domains, will this be incorrectly passes a `same:domain` event?

Reply:
> Could be my lack of previous reviews on this new API, but why is this new event needed in addition to window:load? Or was it meant to replace window:load?

This is utilized by [navigation.js](https://github.com/cypress-io/cypress/blob/ce03f5d86d177ce5cba92016b8f6edf74c163712/packages/driver/src/cy/commands/navigation.ts#L373), since it needs to resolve the command log when the window loads or fails to load. Originally, I had this as 3 events (one of them being `window:load`), but that command log resolution should only happen once, so it was a mess of unsubbing from the other 2 events when one fired. This simplifies it quite a bit.

It's not meant to replace `window:load`. It's only for internal use. Users will still utilize `window:load`, but it's partitioned by domain, since it's not possible to access the `window` object we provide to the event handler since it's cross-domain.

> If there is page navigation window.history.go() or cy.go() between the original & previous domains, will this be incorrectly passes a same:domain event?

Not if it's cross-domain. The [window load event](https://github.com/cypress-io/cypress/blob/ce03f5d86d177ce5cba92016b8f6edf74c163712/packages/driver/src/cypress/cy.ts#L508) won't fire in the primary domain because there will be a cross-origin error. `window:load` isn't currently fired for cross-domain navigation, but that will be implemented in the future.